execution_engine/src/storage/trie/tests.rs

```diff
@@ -92,12 +92,73 @@ mod pointer_block {
 }
 
 mod proptests {
+    use std::convert::TryInto;
+
     use proptest::prelude::*;
 
     use casper_hashing::Digest;
-    use casper_types::{bytesrepr, gens::key_arb, Key, StoredValue};
+    use casper_types::{
+        bytesrepr::{self, deserialize_from_slice, FromBytes, ToBytes},
+        gens::key_arb,
+        Key, StoredValue,
+    };
+
+    use crate::storage::trie::{gens::*, LazilyDeserializedTrie, PointerBlock, Trie};
+
+    fn test_trie_roundtrip_to_lazy_trie<K, V>(trie: &Trie<K, V>, check_key: bool)
```

Comment:
Do we need the `check_key` param here? Or maybe we can infer it below?

`let check_key = matches!(deserialized_from_slice, LazilyDeserializedTrie::Leaf(_))`?

Reply:
Fixed in: https://github.com/casper-network/casper-node/pull/4083/commits/765677f7c07be92bd21cbfd3303dfcf9e74cd819